data-prepper-plugins/geoip-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/geoip/provider/LocationData.java

```diff
@@ -1,43 +1,83 @@
 package com.amazon.dataprepper.plugins.prepper.geoip.provider;
 
-
-// This is a first pass of LocationData, simply to make sure lookup correctly passes through data.
-//TODO Find the best way to attach the location data to a document, and modify this class to allow that.
-
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Objects;
 
 public class LocationData {
+    private final String ip;
     private final String countryName;
     private final String subdivisionName;
     private final String cityName;
+    private final Map<String, Object> location;
+    private final Double latitude;
+    private final Double longitude;
+    private final String continentCode;
+    private final String countryCode;
+    private final String postalCode;
+    private final String regionCode;
+    private final String timezone;
 
-    public LocationData(String countryName, String subdivisionName, String cityName) {
-        this.countryName = countryName;
-        this.subdivisionName = subdivisionName;
-        this.cityName = cityName;
+    public LocationData(Map<Fields, Object> fields) {
```

Comment:
Instead of requiring callers to pass in a map of field keys, it might be better to create a Builder for this class so that this class doesn't get coupled to the `Fields` enum (from previous comment, we might end up with multiple types of enums).

So something that would allow specific Providers to make calls like `LocationData.builder().withCountryName("Canada")` etc.